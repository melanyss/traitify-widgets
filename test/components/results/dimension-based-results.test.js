import renderJSON from "preact-render-to-json";
import Component from "components/results/dimension-based-results";
import {createElement, domHooks} from "support/dom";

jest.mock("lib/with-traitify");
jest.mock("components/results/dimension-based-results/radar", ()=>(()=>(<div className="mock">Radar</div>)));
jest.mock("components/results/dimension-based-results/personality-types", ()=>(()=>(<div className="mock">Personality Types</div>)));
jest.mock("components/results/dimension-based-results/personality-heading", ()=>(()=>(<div className="mock">Personality Heading</div>)));
jest.mock("components/results/dimension-based-results/personality-details", ()=>(()=>(<div className="mock">Personality Details</div>)));
jest.mock("components/results/dimension-based-results/dimensions", ()=>(()=>(<div className="mock">Dimensions</div>)));
jest.mock("components/results/dimension-based-results/personality-traits", ()=>(()=>(<div className="mock">Personality Traits</div>)));

describe("DimensionBasedResults", ()=>{
  domHooks();

  it("renders results", ()=>{
    const tree = renderJSON(<Component />, createElement());

    expect(tree).toMatchSnapshot();
  });
});
