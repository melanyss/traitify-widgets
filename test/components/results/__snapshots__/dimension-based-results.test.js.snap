// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DimensionBasedResults renders results 1`] = `
<section>
  <div
    class="mock"
  >
    Radar
  </div>
  <div
    class="mock"
  >
    Personality Types
  </div>
  <div
    class="mock"
  >
    Personality Heading
  </div>
  <div
    class="mock"
  >
    Personality Details
  </div>
  <div
    class="mock"
  >
    Dimensions
  </div>
  <div
    class="mock"
  >
    Personality Traits
  </div>
</section>
`;
